### PR TITLE
fix(migration):use latest json-io to improve missing field during migration

### PR DESCRIPTION
**Pull request by sgandon**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix

**What is the current behavior?** (You can also link to an open issue here)
when deserializing a field that is missing that was serialized with enough type information, the handler was notified with a null  value

**What is the new behavior?**
Now if the object can be deserialized, the handle gets the proper value. This is useful for field that where renamed.

**Does this PR introduce a breaking change?**

- [X] No

**Other information**:
This PR will require some Studio update for the new json-io version. So please merge this branch https://github.com/Talend/studio-se-master/tree/bugfix/json-io_migration_with_complex_types once this PR is merged.

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/32/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/32/)

**Comment by sgandon** [comment]
@pbailly, @RyanSkraba 
hello guys, could one of you review this please ?

File: pom.xml [hunk 1]
```diff
@@ -86,7 +86,7 @@
             <dependency>
                 <groupId>com.cedarsoftware</groupId>
                 <artifactId>json-io</artifactId>
-                <version>4.5.0</version>
+                <version>4.9.9-TALEND</version>
```
Comment (pbailly) [inline pom.xml]: Is there a reason to not use the default 4.9.9 version?
If there is, could you add a comment on the code describing why? I will be helpful for futures modifications.

Comment (sgandon) [inline pom.xml]: you are absolutely right, I'll do it right now.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/46/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/46/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/101/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/101/)

**Pull request merged**